app/models/audio_version.rb

```diff
@@ -1,26 +1,21 @@
 # encoding: utf-8
 
 class AudioVersion < BaseModel
+  belongs_to :story, -> { with_deleted },
+             class_name: 'Story',
+             foreign_key: 'piece_id', touch: true
+
+  belongs_to :template, -> { with_deleted },
```

Comment:
It may also be convenient to return a link to the audio-version's template in the representer.